docs/services-web-app.md

```diff
@@ -12,3 +29,255 @@ access to private repos and increases API quotas. To set up a personal access to
 1. Select all repo scopes
 1. Copy the `/services/web-app/.env.example` file and rename to `/services/web-app/.env.local`
 1. Add your token to that new `.env.local` file
+
+## Dependencies Set up
+
+### Run Mode
+
+The app uses run-mode to make decisions that are environment-dependant (cookies, next-server...).
+This package configuration is also needed for proper functionality of the Auth Package.
+
+Make sure required environment variables are configured for proper functionality of this package.
+(See [run-mode package](./packages-api.md#run-mode) for details on how to set up the run-mode
+package)
+
+_Note_: this package requires different values depending on environemnt, make sure to update these
+when running on dev/test/production mode
+
+### Auth
+
+This service depends on the auth package; Make sure required environment variables are configured
+for proper functionality of this package. (See [auth package](./packages-api.md#auth) for details on
+how to set up the auth package)
+
+_Note_: this package requires different environment vairables and values depending on environemnt,
+make sure to update these when running on dev/test/production mode
+
+## Adding Local Certificates
+
+The application must run on https for IBMid authentication to work properly. For such purposes,
+local certificates must be generated to run on development environment. You may use
+[mkcert](https://github.com/FiloSottile/mkcert#installation) tool for this. With the tool
+downloaded, the certificates will be automatically generated next time you
+[run the app securely](#running-app-securely)
+
+## Running App Securely
+
+To run the app on https, it must be served from [server.js](../services/web-app/server.js). To do
+this, run the following command from the [web-app's directory](../services/web-app):
+
+`npm run start-secure`
+
+## Protecting a Route
+
+### Protecting Server-Side Rendered Pages
+
+In order to require authentication before a user can access a server-side rendered page use the
+`getPropsWithAuth` utlity function to wrap the pages' _getServerSideProps_ function and wrap the
+pages' content inside the `RequireAuth` component.
+
+#### getPropsWithAuth
+
+`getPropsWithAuth` expects an `authorizationChecker` function that receives the server side context
+as a param and must return a boolean value indicating whether the user is authorized to view the
+content or not:
+
+```
+import { getPropsWithAuth } from '@/utils/getPropsWithAuth'
+import { retrieveUser } from '@/utils/retrieveUser'
+...
+// Your custom authorization logic here, this one considers the user as authorized if it's email address ends in "ibm.com"
+const isValidIbmUser = async (context) => {
+  const user = await retrieveUser(context)
+  if (user) {
+    return user.email?.endsWith('ibm.com')
```

Comment:
I just realized we have a security hole here, lol. If I went and registered `iamnotreallyibm.com`, this check would still pass. So we should adjust this to check for either `.ibm.com` or `@ibm.com` as the suffix.

Reply:
tracked on #210 , fixed on f89271b97dbad6085201d96f8af4f799fedaca2d and updated docs accordingly